repository_service_tuf_api/tasks.py

```diff
@@ -2,24 +2,78 @@
 #
 # SPDX-License-Identifier: MIT
 
-from typing import Any, Optional
+import enum
+from datetime import datetime
+from typing import Any, Optional, Union
 
 from pydantic import BaseModel, Field
 
 from repository_service_tuf_api import repository_metadata
 
 
+class TaskState(str, enum.Enum):
+    PENDING = "PENDING"
+    RECEIVED = "RECEIVED"
+    STARTED = "STARTED"
+    RUNNING = "RUNNING"
+    SUCCESS = "SUCCESS"
+    FAILURE = "FAILURE"
+    REVOKED = "REVOKED"
+    REJECTED = "REJECTED"
+    RETRY = "RETRY"
+    IGNORED = "IGNORED"
+
+
+class TaskName(str, enum.Enum):
+    ADD_TARGETS = "add_targets"
+    REMOVE_TARGETS = "remove_targets"
+    BOOTSTRAP = "bootstrap"
+    UPDATE_SETTINGS = "update_settings"
+    PUBLISH_TARGETS = "publish_targets"
+    METADATA_UPDATE = "metadata_update"
+    SIGN_METADATA = "sign_metadata"
+    DELETE_SIGN_METADATA = "delete_sign_metadata"
+
+
 class GetParameters(BaseModel):
     task_id: str
 
 
+class TaskDetails(BaseModel):
+    message: str = Field(description="Result detail description")
+    error: Optional[str] = Field(
+        description="If task status result is `False` shows error message"
+    )
+    any: Any = Field(description="Any releavant information from task")
+
+
+class TaskResult(BaseModel):
+    status: bool = Field(
+        description="Task result status. `True` Success | `False` Failure"
+    )
+    task: TaskName = Field(description="Task name by worker")
+    last_update: datetime = Field(description="Last time task was updated")
+    details: TaskDetails = Field(description="Relevant result details")
+
+
 class TasksData(BaseModel):
     task_id: str = Field(description="Task ID")
-    state: str = Field(
-        description="State according with Celery Tasks",
+    state: TaskState = Field(
+        description=(
+            "The Celery task state. Note: It isn't the task result status.\n\n"
+            "`PENDING`: Task state is unknown (assumed pending since you know "
+            "the id).\n\n"
+            "`RECEIVED`: Task received by a worker (only used in events).\n\n"
+            "`STARTED`: Task started by a worker.\n\n"
+            "`RUNNING`: Task is running.\n\n"
+            "`SUCCESS`: Task success.\n\n"
```

Comment:
All of the others are in past tense or present continuos tense, so I guess this should be too.
```suggestion
            "`SUCCESS`: Task succeeded.\n\n"
```